experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/lazyField.spec.ts

```diff
@@ -149,314 +148,663 @@ describe("LazyOptionalField", () => {
+			it("With value", () => {
+				const builder = new SchemaBuilder("test", undefined, leafDomain.library);
+				const numberLeafSchema = builder.leaf("number", ValueSchema.Number);
+				const rootSchema = SchemaBuilder.fieldOptional(numberLeafSchema);
+				const schema = builder.intoDocumentSchema(rootSchema);
+
+				const { context, cursor } = initializeTreeWithContent({ schema, initialTree: 42 });
+
+				const field = new LazyOptionalField(
+					context,
+					SchemaBuilder.fieldOptional(numberLeafSchema),
+					cursor,
+					rootFieldAnchor,
+				);
+
+				assert.equal(field.length, 1);
+			});
+		});
+
+		/**
+		 * Creates a tree whose root node contains a single (optional) leaf field.
+		 * Also initializes a cursor and moves that cursor to the tree's root field.
+		 *
+		 * @returns The root node's field.
+		 */
+		function createOptionalLeafTree(
+			kind: ValueSchema,
+			initialTree?:
+				| SchemaAware.TypedField<FieldSchema, SchemaAware.ApiMode.Flexible>
+				| readonly ITreeCursorSynchronous[]
+				| ITreeCursorSynchronous,
+		): LazyOptionalField<[TreeSchema<"leaf">]> {
+			const builder = new SchemaBuilder("test");
+			const leafSchema = builder.leaf("leaf", kind);
+			const rootSchema = SchemaBuilder.fieldOptional(leafSchema);
+			const schema = builder.intoDocumentSchema(rootSchema);
+
+			const { context, cursor } = initializeTreeWithContent({ schema, initialTree });
+
+			return new LazyOptionalField(
+				context,
+				SchemaBuilder.fieldOptional(leafSchema),
+				cursor,
+				rootFieldAnchor,
 			);
+		}
+
+		describe("map", () => {
+			it("No value", () => {
+				const field = createOptionalLeafTree(ValueSchema.Number, undefined);
+
+				assert.deepEqual(
+					field.map((value) => value),
+					[],
+				);
+			});
+
+			it("Primitive field", () => {
+				const field = createOptionalLeafTree(ValueSchema.Boolean, false);
+
+				assert.deepEqual(
+					field.map((value) => value),
+					[false],
+				);
+			});
 		});
 
-		it("Struct", () => {
-			// #region Tree and schema initialization
+		describe("mapBoxed", () => {
+			it("No value", () => {
+				const field = createOptionalLeafTree(ValueSchema.String, undefined);
 
-			const builder = new SchemaBuilder("test");
-			const booleanLeafSchema = builder.leaf("bool", ValueSchema.Boolean);
-			const numberLeafSchema = builder.leaf("number", ValueSchema.Number);
-			const structLeafSchema = builder.struct("struct", {
-				foo: SchemaBuilder.fieldRequired(booleanLeafSchema),
-				bar: SchemaBuilder.fieldOptional(numberLeafSchema),
+				const mapResult = field.mapBoxed((value) => value);
+				assert.deepEqual(mapResult, []);
+			});
+
+			it("With value", () => {
+				const field = createOptionalLeafTree(ValueSchema.Number, 42);
+
+				const mapResult = field.mapBoxed((value) => value);
+				assert.equal(mapResult.length, 1);
+				assert.equal(mapResult[0].value, 42);
 			});
+		});
+	});
+
+	describe("LazyValueField", () => {
+		it("is", () => {
```

Comment:
"is" is implemented on LazyField (as is a lot of other functionality). I don't see any value to testing it for every subclass when they all share an implementation. I recommend making a LazyField set of tests which tests things all LazyFields have in common.

You can do this with a test specific subclass of LazyField, which will help ensure that any future changes which make move these implementation to the subclass will be made aware they need to update the tests (since the test subclass would also need to be updated for such a change)

Reply:
If for some reason you really want to run tests like this over all the different field subclasses (and I don't think we should), you can still write a single general test suite, then run that suite over each subclass.

Reply:
Note that if you want to do black-box testing, you could make a test suite for TreeField, then run that on all the subclasses, and that would be portable to implementations other than LazyField. I don't think the extra abstraction is worth it currently, but its an option which would make sense if we had multiple implementations

Reply:
I definitely agree with that for things like `is` and `parent`, whose behaviors are not impacted by the kinds of contents the subclasses can include. But for things like `map`, the output differs enough that I think separate, easy-to-parse tests are worth the extra boilerplate.